zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1020,6 +1021,21 @@ public static List<ACL> fixupACL(String path, List<Id> authInfo, List<ACL> acls)
                 // authenticated ids of the requestor
                 boolean authIdValid = false;
                 for (Id cid : authInfo) {
+
+                    // Special handling for super user id / cross domain component use cases
+                    // when X509ClientIdAsAcl is enabled
+                    if (X509AuthenticationConfig.getInstance().isX509ClientIdAsAclEnabled() && cid
+                        .getScheme().equals(X509AuthenticationUtil.SUPERUSER_AUTH_SCHEME)) {
+                        authIdValid = true;
+
+                        // Cross domain components
+                        if (!cid.getId().equals(
```

Comment:
Thanks for your comment. Yes this is a problem, and according to our discussion offline this morning, I again changed this part for super users. We will let super user pass and set whatever ACLs they specify in the acl list that's passed in as the last parameter to fixupACL method as the znode ACL, instead of "world, anyone". Please check the updated code and let me know how it looks! :)